src/lib/unpackPlotProps.js

```diff
@@ -29,9 +29,47 @@ export default function unpackPlotProps(props, context) {
   }
 
   let isVisible = false;
+  // Base visibility rules
   if (props.show || (fullValue !== void 0 && fullValue !== null)) {
     isVisible = true;
   }
+  // We can override base visibility rules with a customConfig
+  if (
```

Comment:
I would extract this out to a function somewhere so it's super-clear what the inputs are to the function, and it can be more easily unit-tested.